plugin/src/Plugin/GHCSpecter/Types.hs

```diff
@@ -106,11 +110,25 @@ getMsgQueue =
 assignModuleToDriverId :: DriverId -> ModuleName -> IO ()
 assignModuleToDriverId drvId modName =
   atomically $ do
-    drvModMap <- psDrvIdModuleMap <$> readTVar sessionRef
-    let drvModMap' = insertToBiKeyMap (drvId, modName) drvModMap
+    s <- readTVar sessionRef
+    let drvModMap = psDrvIdModuleMap s
+        drvModMap' = insertToBiKeyMap (drvId, modName) drvModMap
     modifyTVar' sessionRef $ \s -> s {psDrvIdModuleMap = drvModMap'}
 
+assignModuleFileToDriverId :: DriverId -> FilePath -> IO ()
```

Comment:
I think it is better to stay in STM monad for better composability, i.e. use `atomically` at call site.

Reply:
done!